Smolagents: support ImageContent and AudioContent

HSGamer opened the pull request:
Smolagents does have a method to dynamically handle types of tool result (https://github.com/huggingface/smolagents/blob/fc73322658a2c261cf59d817660c6c88d510431b/src/smolagents/agent_types.py#L262-L280). It supports Text as `str`, Image as `PIL.Image` and Audio as `Tensor`.

This PR modified the content to match the supported types.

This is useful for ToolCallingAgent since it supports all types of content. It's not useful for CodeAgent at the moment since it only supports Text, but this can be a preparation for when CodeAgent is upgraded to support all types.

grll commented on src/mcpadapt/smolagents_adapter.py:
```diff
@@ -85,7 +88,7 @@ def __init__(
                 self.skip_forward_signature_validation = True
 
-            def forward(self, *args, **kwargs) -> str:
+            def forward(self, *args, **kwargs):
```
maybe we could type the return type here as image, audio or text

HSGamer replied:
Since `Pillow` and `torchaudio` are optional, I'm not sure that it won't throw error if the packages are not available

grll commented on src/mcpadapt/smolagents_adapter.py:
```diff
@@ -104,12 +107,35 @@ def forward(self, *args, **kwargs) -> str:
+                    if not _is_package_available("torchaudio"):
+                        raise ValueError(
+                            "Audio content requires the torchaudio package to be installed. "
+                            "Please install it with `pip install torchaudio`.",
```
here maybe we should recommend to use the extra dependecy group like:
`uv add mcpadapt[smolagents,audio]` instead of isntalling torchaudio separetly?

grll commented on pyproject.toml:
```diff
@@ -55,6 +55,9 @@ crewai = [
 google-genai = [
     "google-genai>=1.2.0",
 ]
+audio = [
```
we would need to document that if you want to use audio you need to add audio extra, I think raising the error at runtime in the adapter is not sufficient.

grll replied:
Also we would need to add audio to the test group

grll commented on src/mcpadapt/smolagents_adapter.py:
```diff
@@ -104,12 +107,35 @@ def forward(self, *args, **kwargs) -> str:
+                if isinstance(content, mcp.types.TextContent):
+                    return content.text
+
+                if isinstance(content, mcp.types.ImageContent):
```
would be good to have a test for the image scenario

grll commented on src/mcpadapt/smolagents_adapter.py:
```diff
@@ -104,12 +107,35 @@ def forward(self, *args, **kwargs) -> str:
+                    image = Image.open(BytesIO(image_data))
+                    return image
+
+                if isinstance(content, mcp.types.AudioContent):
```
would be good to have a test for the audio scenario

grll commented:
@HSGamer we are almost there. Lint and test are failing though. Could we also maybe simplify the tests and not create the audio or the image everytime but instead maybe create it once and commit it as a file in `tests/data` for example? I think this could cut the code in the test by a significant amount. Thanks a lot for this!

HSGamer commented:
@grll I think I fixed the failing tests, at least `make lint` and `make tests` work on my computer. About the files, I created the sample files and added a module to load them. Can you check again?

grll commented:
> @grll I think I fixed the failing tests, at least `make lint` and `make tests` work on my computer. About the files, I created the sample files and added a module to load them. Can you check again?

thanks for the changes! for some reason the tests still fail in CI, I will have a look tomorrow
